platform/openide.loaders/src/org/openide/text/DataEditorSupport.java

```diff
@@ -1353,19 +1331,8 @@ public void run() throws IOException {
                 try {
                     des.superSaveDoc();
                 } catch (UserQuestionException ex) {
-                    if (GraphicsEnvironment.isHeadless()) {
```

Comment:
This check has been added by Martin Entlicher half a year ago, before it become possible to show `NotifyDescriptor` dialogs in VSCode. Now the dialog could be shown (but we don't want it anyway). As such I am removing the headless check and replacing it with a branding API that allows `java/java.lsp.server/nbcode` suite to explicitly say 'no'.

Reply:
Note that this code was released with 12.2; IMHO in headless environment the code should default to 'yes' setting to remain compatible. Otherwise declare an incompatible change.

Reply:
I'm opting for declaring the change as an incompatible change in this case. The previous change was neither documented or tested and it is there just a three months...

Reply:
Done in 4f42da8398.